libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -505,6 +505,31 @@ bool AP_BattMonitor::get_cycle_count(uint8_t instance, uint16_t &cycles) const
     return drivers[instance]->get_cycle_count(cycles);
 }
 
+bool AP_BattMonitor::get_serial_number(uint8_t instance, char *serial_number, uint8_t buflen) const
+{
+    if (instance >= AP_BATT_MONITOR_MAX_INSTANCES || (drivers[instance] == nullptr)) {
+        return false;
+    }
+    return drivers[instance]->get_serial_number(serial_number, buflen);
+}
+
+bool AP_BattMonitor::get_product_name(uint8_t instance, char *product_name, uint8_t buflen) const
+{
+    if (instance >= AP_BATT_MONITOR_MAX_INSTANCES || (drivers[instance] == nullptr)) {
+        return false;
+    }
+
+    char device_name[SMBUS_READ_BLOCK_MAXIMUM_TRANSFER+1]{}; 
+    char manufacturer_name[SMBUS_READ_BLOCK_MAXIMUM_TRANSFER+1]{};
+
+    bool got_manufacturer_name = drivers[instance]->get_manufacturer_name(manufacturer_name);
+    bool got_device_name = drivers[instance]->get_device_name(device_name);
+
+    snprintf(product_name, buflen, "%s_%s", device_name, manufacturer_name);
+
+    return got_manufacturer_name && got_device_name;
```

Comment:
```suggestion

    char manufacturer_name[SMBUS_READ_BLOCK_MAXIMUM_TRANSFER+1];
    if (!drivers[instance]->get_manufacturer_name(manufacturer_name)) {
        return false;
    }

    char device_name[SMBUS_READ_BLOCK_MAXIMUM_TRANSFER+1]; 
    if (!drivers[instance]->get_device_name(device_name)) {
        return false;
    }

    snprintf(product_name, buflen, "%s_%s", device_name, manufacturer_name);

    return true;
```

Avoids having to initialise the buffers on the stack.  Avoids unnecessary work if we fail to get one of the values.

Should we handle the case we get one or the other but not both, 'though?

Reply:
My thought would be to require both until a case comes up ... extra logic for what one would hope is a nonexistent case... (famous last words)

Or only require device_name as it is gives more info. than manufacturer name alone

Reply:
I made it such that it returns true if either both are available or if one is available but not both